route_generator/src/route_generator.cpp

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (C) 2019 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+#include <ros/ros.h>
+
+#include "route_generator.h"
+
+RouteGenerator::RouteGenerator(int argc, char **argv, const std::string &name) : route_is_active_(false)
+{
+    ros::init(argc, argv, name);
+}
+
+RouteGenerator::~RouteGenerator() {}
+
+void RouteGenerator::initialize()
+{
+    nh_.reset(new ros::CARMANodeHandle);
+    pnh_.reset(new ros::CARMANodeHandle("~"));
+    nh_->getParam("route_file_path", route_file_path_);
+    route_file_path_pub_ = nh_->advertise<std_msgs::String>("selected_route_path", 1);
+    get_available_route_srv_ = nh_->advertiseService("get_available_routes", &RouteGenerator::get_available_route_cb, this);
+    set_active_route_srv_ = nh_->advertiseService("set_active_route", &RouteGenerator::set_active_route_cb, this);
+    start_active_route_srv_ = nh_->advertiseService("start_active_route", &RouteGenerator::start_active_route_cb, this);
+    abort_active_route_srv_ = nh_->advertiseService("abort_active_route", &RouteGenerator::abort_active_route_cb, this);
+}
+
+void RouteGenerator::run()
+{
+    initialize();   
+    ros::CARMANodeHandle::spin();
+}
+
+bool RouteGenerator::get_available_route_cb(cav_srvs::GetAvailableRoutesRequest &req, cav_srvs::GetAvailableRoutesResponse &resp)
+{
+    ROS_ERROR_STREAM("The call back is triggered!");
+    std::vector<std::string> route_ids = read_route_names(route_file_path_);
+    ROS_ERROR_STREAM("Got a list of names!");
+    ROS_ERROR_STREAM(route_ids.size());
+    for(int i = 0; i < route_ids.size(); ++i)
+    {
+        std::string route_name = route_ids[i].substr(0, route_ids[i].find("."));
```

Comment:
This looks to me like it should brake if you had a directory such as 
/opt/carma/.hidden_routes/

Can you make it check for ".csv" instead?

Reply:
Make sense. Changed.